### PR TITLE
Test suite & fixes for validate package

### PR DESCRIPTION
**Pull request by twitchyliquid64**
I've started working through the [official test suite](https://github.com/WebAssembly/spec/tree/master/test/core) and representing the tests specified in there as tests on the validate module.

As I go, I'm fixing any issues I find.

### CODE REVIEW

**Comment by twitchyliquid64** [comment]
Okay, I think thats a good amount of tests for one PR. Feel free to review at your leisure (no rush :smile: ).

File: validate/validate_test.go [hunk 1]
```diff
@@ -0,0 +1,1203 @@
+package validate
```
Comment (sbinet) [inline validate/validate_test.go]: please add a license header

File: validate/validate_test.go [hunk 2]
```diff
@@ -0,0 +1,1203 @@
...
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			mod := wasm.Module{}
+			sig := wasm.FunctionSig{Form: 0x60 /* Must always be 0x60 */}
+			fn := wasm.FunctionBody{Module: &mod, Code: tc.code}
+
+			_, err := verifyBody(&sig, &fn, &mod)
+			if err != tc.err {
+				t.Errorf("verify returned '%v', want '%v'", err, tc.err)
```
Comment (sbinet) [inline validate/validate_test.go]:
```suggestion
				t.Fatalf("verify returned '%v', want '%v'", err, tc.err)
```
(it's in a `testing.T.Run` scope)

File: validate/validate_test.go [hunk 3]
```diff
@@ -0,0 +1,1203 @@
...
```
Comment (sbinet) [inline validate/validate_test.go]:
```suggestion
				t.Fatalf("verify returned '%v', want '%v'", err, tc.err)
```

File: validate/validate_test.go [hunk 4]
```diff
@@ -0,0 +1,1203 @@
...
+				Module: &mod,
+				Code:   tc.code,
+				Locals: []wasm.LocalEntry{
+					{Count: 1, Type: wasm.ValueTypeI32},
+				},
+			}
+
+			_, err := verifyBody(&sig, &fn, &mod)
+			if err != tc.err {
+				t.Errorf("verify returned '%v', want '%v'", err, tc.err)
```
Comment (sbinet) [inline validate/validate_test.go]:
```suggestion
				t.Fatalf("verify returned '%v', want '%v'", err, tc.err)
```

File: validate/validate_test.go [hunk 5]
```diff
@@ -0,0 +1,1203 @@
...
+				Locals: []wasm.LocalEntry{
+					{Count: 1, Type: wasm.ValueTypeI32},
+					{Count: 1, Type: wasm.ValueTypeI64},
+					{Count: 1, Type: wasm.ValueTypeF64},
+				},
+			}
+
+			_, err := verifyBody(&sig, &fn, &mod)
+			if err != tc.err {
+				t.Errorf("verify returned '%v', want '%v'", err, tc.err)
```
Comment (sbinet) [inline validate/validate_test.go]:
```suggestion
				t.Fatalf("verify returned '%v', want '%v'", err, tc.err)
```

File: validate/validate_test.go [hunk 6]
```diff
@@ -0,0 +1,1203 @@
...
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			mod := wasm.Module{}
+			sig := wasm.FunctionSig{Form: 0x60 /* Must always be 0x60 */}
+			fn := wasm.FunctionBody{Module: &mod, Code: tc.code}
+
+			_, err := verifyBody(&sig, &fn, &mod)
+			if err != tc.err {
+				t.Errorf("verify returned '%v', want '%v'", err, tc.err)
```
Comment (sbinet) [inline validate/validate_test.go]:
```suggestion
				t.Fatalf("verify returned '%v', want '%v'", err, tc.err)
```

File: validate/validate_test.go [hunk 7]
```diff
@@ -0,0 +1,1203 @@
...
+					*mod.FunctionIndexSpace[1].Sig,
+				},
+			}
+
+			sig := wasm.FunctionSig{Form: 0x60 /* Must always be 0x60 */}
+			fn := wasm.FunctionBody{Module: &mod, Code: tc.code}
+
+			_, err := verifyBody(&sig, &fn, &mod)
+			if err != tc.err {
+				t.Errorf("verify returned '%v', want '%v'", err, tc.err)
```
Comment (sbinet) [inline validate/validate_test.go]:
```suggestion
				t.Fatalf("verify returned '%v', want '%v'", err, tc.err)
```

**Comment by twitchyliquid64** [comment]
@sbinet this is ready, but the two new PR checks are failing ... what are they? can we skip them?

**Comment by sbinet** [comment]
yeah, codecov is being a bit too stringent (and speaks up when the code coverage drops by some tiny margin.)
not sure how to change that...

**Pull request merged**